src/stdio_mgr/types.py

```diff
@@ -0,0 +1,175 @@
+r"""``stdio_mgr.types`` *code module*.
+
+``stdio_mgr.types`` provides context managers for convenient
+interaction with ``stdin``/``stdout``/``stderr`` as a tuple.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The MIT License; see |license_txt|_ for full license terms
+
+**Members**
+
+"""
+import collections.abc
+from contextlib import suppress
+from io import TextIOBase
+
+from stdio_mgr.compat import AbstractContextManager
+
+
+class MultiItemTuple(collections.abc.Iterable):
+    """Iterable with methods that operate on all items."""
+
+    @staticmethod
+    def _invoke_name(item, method_name):
+        item = getattr(item, method_name)
+        if callable(item):
+            item = item()
+        return item
```

Comment:
And then returns either the initial noncallable thing plucked off (which should NEVER be the case for a method, b/c methods are always(?) callable) or the result of the zero-argument invocation of the thing plucked off.